### DISCUSSION TITLE
`mutate` signature for optimistic updates?

### DISCUSSION THREAD

**Discussion by SevenOutman** [vercel/swr#1721] [Ideas]
While there's example of making optimistic update with two separate `mutate` calls, why not add a new `mutate` signature to act as a shortcut?

```js
mutate(newUser, updateUser(newUser), false)

// Equivalent to

mutate(newUser, false)
mutate(updateUser(newUser), false)
```

**Comment by shuding** [comment]
Related to #1729, an option can be added to make it available:

```js
mutate(updateUser(newUser), {
  optimisticData: newUser
})
```

I think in general, for advanced use cases explicit option names are better than conventions (the order of parameters) for people to remember.

(Not sure about the name `optimisticData` though.)

**Reply by SevenOutman** [reply]
> I think in general, for advanced use cases explicit option names are better than conventions (the order of parameters) for people to remember.

That makes sense to me.

**Reply by shuding** [reply]
Would be great if you can give #1745 a look!
